subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -559,7 +559,11 @@ static void mqtt_evt_handler(struct mqtt_client *const c,
 			break;
 		}
 
-		if (!mqtt_evt->param.connack.session_present_flag) {
+		bool session_present = false;
```

Comment:
I'm actually a bit confused about this change, have you observed that some MQTT broker returned session present flag set to 1 even if clean session was requested? It's against MQTT spec to do so:
> If the Server accepts a connection with CleanSession set to 1, the Server MUST set Session Present to 0 in the CONNACK packet

Reply:
But if it really happens, IMO it would be cleaner to rely on a runtime configuration:
```
session_present = c->clean_session ? 0 : mqtt_evt->param.connack.session_present_flag;
```

Reply:
@rlubos it does happen on AWS IoT -- they are violating that rule.